lib/debug/reflection.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module DEBUGGER__
+  module Reflection
+    module_function def instance_variables_of(o)
+      M_INSTANCE_VARIABLES.bind_call(o)
+    end
```

Comment:
Connecting the debugger to a Ruby version pre-2.7 will have the effect of back-porting the `bind_call` method, which would otherwise exist.

This is convenient for the implementation of the `Ruby/debug` itself, but perhaps it's not a good idea, as developers' code can call `bind_call` successfully, but only when they're debugging, which could be quite confusing.

Although it's more tedious, perhaps it's a better idea to conditionally define one of two implementations of these helpers, rather than polluting `UnboundMethod`? E.g.

```suggestion
    if UnboundMethod.method_defined?(:bind_call)
      module_function def instance_variables_of(o)
        M_INSTANCE_VARIABLES.bind_call(o)
      end
    else
      module_function def instance_variables_of(o)
        M_INSTANCE_VARIABLES.bind(o).call
      end
    end
```

Reply:
I'm not sure we should support pre-2.7 though...

Reply:
If we don't want to support pre-2.7, then we should remove the monkey patch entirely.

I don't personally use anything pre-3.2, but it feels too soon to drop 2.7. In any case, the current implementation of it is incorrect, and we should not leave it as-is.